app/lib/tool/utils/http.dart

```diff
@@ -67,3 +67,24 @@ class _AuthenticatedClient extends http.BaseClient {
     super.close();
   }
 }
+
+/// Allows the call of [close] multiple times without throwing an Exception.
+class _MultiCloseClient extends http.BaseClient {
```

Comment:
So the counter argument to doing this is that now you risk one context closing the client used in another context.

When creating an instance of `_AuthenticatedClient` we can specify `_closeInnerClient = false` in the constructor.

Reply:
Hm, I'm not sure if it is the same to not close the inner client. I agree, ideally, this shouldn't be necessary, but this seems to be slightly better than the obscure failure we were getting.